libbeat/management/status/status.go

```diff
@@ -0,0 +1,55 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package status
+
+// Status describes the current status of the beat.
+type Status int
+
+//go:generate stringer -type=Status
+const (
+	// Unknown is initial status when none has been reported.
+	Unknown Status = iota
+	// Starting is status describing application is starting.
+	Starting
+	// Configuring is status describing application is configuring.
+	Configuring
+	// Running is status describing application is running.
+	Running
+	// Degraded is status describing application is degraded.
+	Degraded
+	// Failed is status describing application is failed. This status should
+	// only be used in the case the beat should stop running as the failure
+	// cannot be recovered.
+	Failed
+	// Stopping is status describing application is stopping.
+	Stopping
+	// Stopped is status describing application is stopped.
+	Stopped
+)
```

Comment:
just a reminder these const values here came from [management.go](https://github.com/elastic/beats/pull/39209/files#diff-fcf0ac1927a6e4a560125bca3691cb6c27006277664735a58500420535fedc27L29) as they had to get to a separate package to avoid cyclic imports between reload.go and management.go. So I am not the original formulator of application vs input here. Now if you ask my opinion this is neither an application nor an input; I think it should be unit as these statuses can apply for both inputs and outputs, but no strong feelings

Reply:
I followed my instincts and went with `unit` 🙂 If you have any objection feel free to un-resolve this